components/org.wso2.micro.integrator.extensions/org.wso2.micro.integrator.management.apis/src/main/java/org/wso2/micro/integrator/management/apis/LogFilesResource.java

```diff
@@ -0,0 +1,173 @@
+        Utils.setJsonPayLoad(axis2MessageContext, jsonBody);
+        axis2MessageContext.removeProperty(NO_ENTITY_BODY);
+    }
+
+    private void populateFileContent(MessageContext synCtx, String pathParameter) {
+
+        org.apache.axis2.context.MessageContext axis2MessageContext =
+                ((Axis2MessageContext) synCtx).getAxis2MessageContext();
+
+        DataHandler dataHandler = downloadArchivedLogFiles(pathParameter);
+        if (dataHandler != null) {
+            try {
+                InputStream fileInput = dataHandler.getInputStream();
+                SOAPFactory factory = OMAbstractFactory.getSOAP12Factory();
+                SOAPEnvelope env = factory.getDefaultEnvelope();
+                OMNamespace ns =
+                        factory.createOMNamespace(RelayConstants.BINARY_CONTENT_QNAME.getNamespaceURI(), "ns");
+                OMElement omEle = factory.createOMElement(RelayConstants.BINARY_CONTENT_QNAME.getLocalPart(), ns);
+                StreamingOnRequestDataSource ds = new StreamingOnRequestDataSource(fileInput);
+                dataHandler = new DataHandler(ds);
+                OMText textData = factory.createOMText(dataHandler, true);
+                omEle.addChild(textData);
+                env.getBody().addChild(omEle);
+                synCtx.setEnvelope(env);
+                axis2MessageContext.removeProperty(NO_ENTITY_BODY);
+                axis2MessageContext.setProperty(Constants.MESSAGE_TYPE, "application/octet-stream");
+                axis2MessageContext.setProperty(Constants.CONTENT_TYPE, "application/txt");
+            } catch (AxisFault e) {
+                log.error("Error occurred while creating the response", e);
+                sendFaultResponse(axis2MessageContext);
+            } catch (IOException e) {
+                log.error("Error occurred while reading the input stream", e);
+                sendFaultResponse(axis2MessageContext);
+            }
+        } else {
+            sendFaultResponse(axis2MessageContext);
+        }
+    }
+
+    private void sendFaultResponse(org.apache.axis2.context.MessageContext axis2MessageContext) {
+
+        axis2MessageContext.setProperty(NO_ENTITY_BODY, true);
+        axis2MessageContext.setProperty(HTTP_SC, 500);
+    }
+
+    private DataHandler downloadArchivedLogFiles(String logFile) {
+
+        InputStream is = null;
+        ByteArrayDataSource bytArrayDS;
+        try {
+            Path logFilePath = Paths.get(Utils.getCarbonLogsPath(), logFile);
+            File file = new File(logFilePath.toString());
+            if (file.exists() && !file.isDirectory()) {
+                is = new BufferedInputStream(new FileInputStream(logFilePath.toString()));
+                bytArrayDS = new ByteArrayDataSource(is, "text/xml");
+                return new DataHandler(bytArrayDS);
+            } else {
+                log.error("Could not find the requested file : " + logFile);
```

Comment:
Shall we print the complete 'logFilePath'

Reply:
+1 fixed